pkg/auth/auth.go

```diff
@@ -0,0 +1,190 @@
+package auth
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/pkg/docker/config"
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// GetDefaultAuthFile returns env value REGISTRY_AUTH_FILE as default --authfile path
+// used in multiple --authfile flag definitions
+func GetDefaultAuthFile() string {
+	return os.Getenv("REGISTRY_AUTH_FILE")
+}
+
+// CheckAuthFile validates filepath given by --authfile
+// used by command has --authfile flag
+func CheckAuthFile(authfile string) error {
+	if authfile == "" {
+		return nil
+	}
+	if _, err := os.Stat(authfile); err != nil {
+		return errors.Wrapf(err, "error checking authfile path %s", authfile)
+	}
+	return nil
+}
+
+// GetCredsAndLogin login to the server with creds from Stdin or CLI
+func GetCredsAndLogin(ctx context.Context, systemContext *types.SystemContext, iopts *LoginOptions, registry string, getLoginSet bool) error {
```

Comment:
What does “iopts” mean, BTW? Should Go capitalization be `iOpts`?

(I don’t know what that means, but it seems simpler to me to use `opts` than to explain the meaning.)